File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    count_tangerine={}
    for i in tangerine:
        if i in count_tangerine:
            count_tangerine[i] +=1
        else: 
            count_tangerine[i] = 1

    count_tangerine = sorted(count_tangerine.items(), key = lambda x: x[1], reverse = True)
    
    answer = 0
    for i in count_tangerine:
        if k <= 0:
            break
        k -= i[1]
        answer += 1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }

        List<Integer> frequencies = new ArrayList<>(map.values());
        frequencies.sort((a, b) -> b - a);

        int kind = 0;
        for (int i = 0; i < frequencies.size(); i++) {
            if (k <= 0) {
                break;
            }

            k -= frequencies.get(i);
            kind++;
        }

        return kind;
    }
}
